assets/js/components/data/index.js

```diff
@@ -255,10 +226,48 @@ const dataAPI = {
 			// Resolve any returned data requests, then re-request the remainder after a pause.
 		} ).catch( ( err ) => {
 			// Handle the error and give up trying.
-			console.log( 'error', err ); // eslint-disable-line no-console
+			console.warn( 'Error caught during combinedGet', err ); // eslint-disable-line no-console
 		} );
 	},
 
+	handleWpError( error ) {
+		const { code, data } = error;
+
+		if ( ! code || ! data ) {
+			return;
+		}
+
+		// eslint-disable-next-line no-console
+		console.warn( 'WP Error in data response', error );
+
+		let addedNoticeCount = 0;
+
+		// Add insufficient scopes warning.
+		if ( [ 'authError', 'insufficientPermissions' ].includes( data.reason ) ) {
+			addFilter( 'googlesitekit.ErrorNotification',
+				'googlesitekit.AuthNotification',
+				fillFilterWithComponent( DashboardAuthAlert ), 1 );
+			addedNoticeCount++;
+		}
+
+		// Insufficient access permissions.
+		if ( 'forbidden' === data.reason ) {
+			addFilter( 'googlesitekit.ErrorNotification',
+				'googlesitekit.AuthNotification',
+				fillFilterWithComponent( DashboardPermissionAlert ), 1 );
+			addedNoticeCount++;
+		}
+
+		if ( addedNoticeCount ) {
+			addFilter( 'googlesitekit.TotalNotifications',
+				'googlesitekit.AuthCountIncrease', ( count ) => {
+					// Only run once.
+					removeFilter( 'googlesitekit.TotalNotifications', 'googlesitekit.AuthCountIncrease' );
+					return count + addedNoticeCount;
+				} );
+		}
```

Comment:
I wonder whether we can somehow act differently if that same notification is already displayed. For example, if 10 responses come back with this, we would currently increase the counter by 10 as far as I can tell, which is not right because we'd still only display the notification once. This is not a super high priority, but if you have an idea, let's discuss whether it's simple enough to include here.

Reply:
I'm not sure the count would be incremented by 10 in this case since each response would add its own filter which would increment the count by 1. However, the first filter to run would remove them all, since they all share the same namespace. It's a bit strange though.

A better approach as an intermediary to using a data store might be to simply set a global flag as to whether or not each notice has been added and then move the count filters out and base those on the flags rather than hooking in each time the components are added. I'd be interested to get @tofumatt 's perspective on this as well.